doc/manual/packages.rst

```diff
@@ -309,6 +309,47 @@ Once you do this, the package manager knows your GitHub user name and can config
 You should also `upload <https://github.com/settings/ssh>`_ your public SSH key to GitHub and set up an `SSH agent <http://linux.die.net/man/1/ssh-agent>`_ on your development machine so that you can push changes with minimal hassle.
 In the future, we will make this system extensible and support other common git hosting options like `BitBucket <https://bitbucket.org>`_ and allow developers to choose their favorite.
 
+Guidelines for Naming a Package
+-------------------------------
+
+Packages should be given names that are understandable to the general Julia
+user population, *even to users who are not domain experts*.
+
+1. Don't use an acronym unless there is minimal possibility of confusion. Avoid
+   jargon.
+
+  * It's ok to say ``USA`` if you're talking about the USA
+  * It's not ok to say ``PMA``, even if you're talking about positive mental attitude
+
+2. Package names should be plural, especially if you want to define a type in
+   your package with a similar name.
+
+  * ``DataFrames`` provides the ``DataFrame`` type.
+  * ``BloomFilters`` provides the ``BloomFilter`` type.
+
+3. Err on the side of clarity, even if clarity seems long-winded to you.
+
+  * If you're aiming to write the definitive Julia tool for data visualization,
```

Comment:
The reason for the name change from CVX to Convex was on request from the author of the CVX matlab package, not for increased clarity. I'd also claim that Convex.jl is one particular way to model convex optimization and isn't the "unique source of truth", so I'm not sure it's a good example.